keras_cv/models/segmentation/segment_anything/sam_test.py

```diff
@@ -0,0 +1,377 @@
+        prompt_encoder_outputs = self.prompt_encoder(
+            dict(points=points, labels=labels, box=box, mask=input_mask)
+        )
+        sparse_embeddings, dense_embeddings, dense_positional_embeddings = (
+            prompt_encoder_outputs["sparse_embeddings"],
+            prompt_encoder_outputs["dense_embeddings"],
+            prompt_encoder_outputs["dense_positional_embeddings"],
+        )
+        image_embeddings = np.random.randn(1, 64, 64, 256)
+        outputs = self.mask_decoder(
+            dict(
+                image_embeddings=image_embeddings,
+                image_pe=dense_positional_embeddings,
+                sparse_prompt_embeddings=sparse_embeddings,
+                dense_prompt_embeddings=dense_embeddings,
+            )
+        )
+        masks, iou_pred = outputs["masks"], outputs["iou_pred"]
+        num_parameters = np.sum(
+            [np.prod(x.shape) for x in self.mask_decoder.weights]
+        )
+        masks, iou_pred = map(ops.convert_to_numpy, [masks, iou_pred])
+        self.assertEqual(masks.shape, (1, 4, 256, 256))
+        self.assertEqual(iou_pred.shape, (1, 4))
+        self.assertEqual(num_parameters, 4_058_340)
+
+    @pytest.mark.large
+    def test_end_to_end_model_predict(self):
+        model = SegmentAnythingModel(
+            backbone=self.image_encoder,
+            prompt_encoder=self.prompt_encoder,
+            mask_decoder=self.mask_decoder,
+        )
+
+        points, labels, box, input_mask = self.get_prompts(1)
+
+        inputs = {
+            "images": np.ones((1, 1024, 1024, 3)),
+            "points": points,
+            "labels": labels,
+            "box": box,
+            "mask": input_mask,
+        }
+
+        # Check the number of parameters
+        num_parameters = np.sum([np.prod(x.shape) for x in model.weights])
+        self.assertEqual(num_parameters, 89_670_912 + 6_476 + 4_058_340)
+
+        # Forward pass through the model
+        outputs = model.predict(inputs)
+        masks, iou_pred = outputs["masks"], outputs["iou_pred"]
+
+        # Check the output is equal to the one we expect if we
+        # run each component separately. This is to confirm that
+        # the graph is getting compiled correctly i.e. the jitted
+        # execution is equivalent to the eager execution.
+        features = self.image_encoder(inputs["images"])
+        outputs_ex = self.prompt_encoder(
+            {k: v for k, v in inputs.items() if k != "images"}
+        )
+        outputs_ex = self.mask_decoder(
+            {
+                "image_embeddings": features,
+                "image_pe": outputs_ex["dense_positional_embeddings"],
+                "sparse_prompt_embeddings": outputs_ex["sparse_embeddings"],
+                "dense_prompt_embeddings": outputs_ex["dense_embeddings"],
+            },
+        )
+        masks_ex, iou_pred_ex = outputs_ex["masks"], outputs_ex["iou_pred"]
+
+        self.assertAllClose(masks, masks_ex, atol=1e-5)
```

Comment:
Looks like for the TF + JAX backends this tolerance needs to be more lie 3e-5

Reply:
Done, thanks!